### PR TITLE
Auto-take profit validations

### PR DESCRIPTION
**Pull request by piekczyk**
# [Validation | The ATP Trigger price cannot be < OSM Current Price](https://app.shortcut.com/oazo-apps/story/6265/validation-the-atp-trigger-price-cannot-be-osm-current-price)

# [Warning | The ATP Trigger price cannot have estimated Col. Ratio < Auto Buy Trigger Col Ratio](https://app.shortcut.com/oazo-apps/story/6297/warning-the-atp-trigger-price-cannot-have-estimated-col-ratio-auto-buy-trigger-col-ratio)

# [Warning | The ATP Trigger price cannot have estimated Col. Ratio < Constant Mulitple Trigger Buy Trigger Col Ratio](https://app.shortcut.com/oazo-apps/story/6298/warning-the-atp-trigger-price-cannot-have-estimated-col-ratio-constant-mulitple-trigger-buy-trigger-col-ratio)

# [Warning | Auto Buy Trigger Col Ratio > estimated ATP Trigger price](https://app.shortcut.com/oazo-apps/story/6300/warning-auto-buy-trigger-col-ratio-estimated-atp-trigger-price)

# [Warning | Constant Mulitple Trigger Buy Trigger Col Ratio > The ATP Trigger estimated Col. Ratio](https://app.shortcut.com/oazo-apps/story/6299/warning-constant-mulitple-trigger-buy-trigger-col-ratio-the-atp-trigger-estimated-col-ratio)

<please insert a shortcut link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>

- added 3x validations to auto-take form (error - immediately execution, warnings - ATP below auto-buy or constant-multiple buy trigger)
- new validation in auto-buy form (warning when auto-buy trigger is above ATP trigger)
- new validation in constant-multiple form (warning when constant multiple buy trigger is above ATP trigger)
  
## How to test 🧪
  <Please explain how to test your changes>

- prepare vault & form states to generate warnings described above

### CODE REVIEW

File: features/automation/optimization/autoTakeProfit/validators.test.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    const warnings = warningsAutoTakeProfitValidation({
+      ...autoTakeProfitWarningsValidationBaseData,
+      ethBalance: new BigNumber(0.0001),
+      autoBuyTriggerPrice: new BigNumber(1400),
+      isAutoBuyEnabled: true,
+    })
+
+    expect(warnings).to.be.deep.eq(['potentialInsufficientEthFundsForTx'])
+  })
+  it('should show warning that ATP is lower than auto-buy trigger', () => {
```
Comment (johnnieskywalker) [inline features/automation/optimization/autoTakeProfit/validators.test.ts:38]: Great job on tests, but I guess for this one and next one `should show warning that ATP is lower than constant-multiple buy trigger` we could also include edge case when ATP is equal auto-buy trigger I guess it should also be displayed in this case

**Review by piotrkonowrocki — DISMISSED** [review]

File: public/locales/en/common.json [hunk 2]
```diff
@@ -939,7 +940,11 @@
...
     "adding-constant-multiple-when-auto-sell-or-buy-enabled": "Creating a Constant multiple will remove your existing {{enabledTriggers}}. Are you sure you want to continue?",
     "constant-multiple-auto-sell-triggered-immediately": "Setting your Sell Trigger at this ratio would result in it being executed immediately",
-    "constant-multiple-auto-buy-triggered-immediately": "Setting your Buy Trigger at this ratio would result in it being executed immediately"
+    "constant-multiple-auto-buy-triggered-immediately": "Setting your Buy Trigger at this ratio would result in it being executed immediately",
+    "auto-take-profit-trigger-lower-than-auto-buy-trigger": "Setting your Target Price at this level may result in your Auto-Buy not executing. Find out more <0>here</0>.",
+    "auto-take-profit-trigger-lower-than-constant-multiple-buy-trigger": "Setting your Target Price at this level may result in your Constant Multiple Buy Trigger not executing. Find out more <0>here</0>.",
```
Comment (johnnieskywalker) [inline public/locales/en/common.json:945]: image

**Review by johnnieskywalker — APPROVED** [review]
